legion/engine/physics/mesh_splitter_utils/mesh_half_edge.hpp

```diff
@@ -0,0 +1,268 @@
+            else if (nextDistFromPlane < -constants::polygonSplitterEpsilon)
+            {
+                y = -1;
+            }
+
+            int edgeSplitState = x * y;
+
+
+            return edgeSplitState < 0;
+
+
+        }
+
+        math::vec3 GetWorldCentroid(const math::mat4& transform) const
+        {
+            return transform * math::vec4((position + nextEdge->position) * 0.5f, 1);
+        }
+
+        bool isEdgePartlyAbovePlane(const math::mat4& transform, const math::vec3& planePosition, const math::vec3& planeNormal)
+        {
+            auto [currentDistFromPlane, nextDistFromPlane] = GetEdgeDistancesFromPlane(transform, planePosition, planeNormal);
+
+            bool currentAbovePlane = currentDistFromPlane > constants::polygonSplitterEpsilon;
+            bool nextAbovePlane = nextDistFromPlane > constants::polygonSplitterEpsilon;
+
+            return currentAbovePlane || nextAbovePlane;
+        }
+
+        bool isEdgePartlyBelowPlane(const math::mat4& transform, const math::vec3& planePosition, const math::vec3& planeNormal)
+        {
+            auto [currentDistFromPlane, nextDistFromPlane] = GetEdgeDistancesFromPlane(transform, planePosition, planeNormal);
+
+            bool currentBelowPlane = currentDistFromPlane < -constants::polygonSplitterEpsilon;
+            bool nextBelowPlane = nextDistFromPlane < -constants::polygonSplitterEpsilon;
+
+            return currentBelowPlane || nextBelowPlane;
+        }
+
+        std::tuple<float, float> GetEdgeDistancesFromPlane(const math::mat4& transform, const math::vec3& planePosition, const math::vec3& planeNormal)
+        {
+            auto [currentWorldPos, nextWorldPos] = GetEdgeWorldPositions(transform);
+
+            float currentDistFromPlane = PhysicsStatics::PointDistanceToPlane(planeNormal, planePosition,  currentWorldPos);
+            float nextDistFromPlane = PhysicsStatics::PointDistanceToPlane(planeNormal, planePosition,  nextWorldPos);
+
+            return std::make_tuple(currentDistFromPlane, nextDistFromPlane);
+        }
+
+        std::tuple<math::vec3, math::vec3> GetEdgeWorldPositions(const math::mat4& transform)
+        {
+            math::vec3 currentWorldPos = GetEdgeWorldPosition(transform);
+            math::vec3 nextWorldPos = nextEdge->GetEdgeWorldPosition(transform);
+
+            return std::make_tuple(currentWorldPos, nextWorldPos);
+        }
+
+        math::vec3 GetEdgeWorldPosition(const math::mat4& transform)
+        {
+            return transform * math::vec4(position, 1);
+        }
+
+
+
+        bool AttemptGetTrianglesInEdges
+        (std::shared_ptr<MeshHalfEdge>& nextEdge, std::shared_ptr<MeshHalfEdge>& prevEdge)
+        {
+            if (!this->nextEdge)
+            {
+                return false;
+            }
+
+            nextEdge = this->nextEdge;
+
+            if (!nextEdge->nextEdge)
+            {
+                return false;
+            }
+
+            prevEdge = nextEdge->nextEdge;
```

Comment:
The memory address where prevEdge points to might change, comment on this. Or keep a temporary variable for this